src/main/java/seedu/address/logic/commands/FindPatientCommand.java

```diff
@@ -0,0 +1,58 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+
+import seedu.address.commons.util.ToStringBuilder;
+import seedu.address.logic.Messages;
+import seedu.address.model.NewModel;
+import seedu.address.model.person.NameContainsKeywordsPatientPredicate;
+
+/**
+ * Finds and lists all patients in database whose name contains any of the argument keywords.
+ * Keyword matching is case insensitive.
+ */
+public class FindPatientCommand extends NewCommand {
+
+    public static final String COMMAND_WORD = "find_p";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Finds all patients whose names contain any of "
+            + "the specified keywords (case-insensitive) and displays them as a list with index numbers.\n"
+            + "Parameters: KEYWORD [MORE_KEYWORDS]...\n"
+            + "Example: " + COMMAND_WORD + " alice bob charlie";
+
+    private final NameContainsKeywordsPatientPredicate predicate;
+
+    public FindPatientCommand(NameContainsKeywordsPatientPredicate predicate) {
+        this.predicate = predicate;
+    }
+
+    @Override
+    public CommandResult execute(NewModel model) {
+        requireNonNull(model);
+        model.updateFilteredPatientList(predicate);
+        return new CommandResult(
+                String.format(Messages.MESSAGE_PATIENTS_LISTED_OVERVIEW, model.getFilteredPatientList().size()));
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        if (other == this) {
+            return true;
+        }
+
+        // instanceof handles nulls
+        if (!(other instanceof FindPatientCommand)) {
+            return false;
+        }
+
+        FindPatientCommand otherFindCommand = (FindPatientCommand) other;
+        return predicate.equals(otherFindCommand.predicate);
+    }
+
```

Comment:
I think @derekjxtan  mentioned add hashcode for wherever there's equals.
Also add tests for hashcode too. can refer to mine.